### DISCUSSION TITLE
How to split database code into multiple files when using transactions

### DISCUSSION THREAD

**Discussion by felher** [com-lihaoyi/scalasql#26] [General]
Hi there,

I was wondering whats the best way to split the database code into multiple files.

I'm currently doing it like this:

```scala
import scalasql.*
import SqliteDialect.*

final case class Counter[T[_]](
    name: T[String],
    count: T[Int]
)

object Counter extends scalasql.Table[Counter]:
  def getOrElse(txn: DbApi.Txn, name: String, default: Int): Int =
    txn.run(
      this.select
        .filter(_.name === name)
        .map(_.count)
        .single
    )
```

and then using it somewhere else by doing:
```scala
sqliteClient.transaction: db =>
  val counterValue = counters.Counter.getOrElse(db, "counter", 0)
  ... more stuff here
```

am I doing this right? It feels a bit odd that I have to `import SqliteDialect.* ` in every file, since it also compiles fine if I mismatch those dialects. I.e. if one of the files imports sqlite and the other postgres, for example.

Kind regards,
Felix

**Comment by lihaoyi** [comment]
You're doing it right. You do need to `import SqliteDialect.*` in every file. We can't guard against everything at compile time, though if you have ideas of how to improve the strictness we can discuss
